pkg/apiclient/apiclient.go

```diff
@@ -67,6 +72,42 @@ func NewClientOrDie(opts *ClientOptions) ServerClient {
 	return client
 }
 
+// JwtCredentials holds a token for authentication.
+type jwtCredentials struct {
+	Token string
+}
+
+func (c jwtCredentials) RequireTransportSecurity() bool {
+	return false
+}
+
+func (c jwtCredentials) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{
+		"tokens": c.Token,
+	}, nil
+}
+
+// endpointCredentials retrieves from configuration the login token for a given endpoint.
+func endpointCredentials(endpoint string) jwtCredentials {
+	credentials := jwtCredentials{}
+
+	localConfig, err := config_util.ReadLocalConfig()
```

Comment:
I won't gate this checkin, but I think we need to move the reading/parsing of the local config, tobe done when creating the API client. A use case is that I should be able to create an argocd client without having an .argocd directory. In this case I would supply an API token when instantiating the client.